crates/gen/src/method.rs

```diff
@@ -235,8 +235,11 @@ impl Method {
         }
     }
 
-    pub fn gen_upcall(&self, inner: TokenStream) -> TokenStream {
-        let invoke_args = self.params.iter().map(|param| param.gen_invoke_arg());
+    pub fn gen_upcall(&self, inner: TokenStream, relative: bool) -> TokenStream {
```

Comment:
Nit: you might want to consider changing the bool to an enum. Something like this:
```rust
enum CallPosition {
  Relative,
  Absolute
}
```
This can make the code easier to read.

Reply:
Yep, good call. Was meaning to do something like this.